tcms/testplans/tests/test_new_plan.py

```diff
@@ -82,10 +111,11 @@ def _test_plan_create_new(self, is_active):
         self.assertEqual(response.status_code, HTTPStatus.FOUND)
 
         plan = TestPlan.objects.get(
-            name=self.plan_name,
+            name=self.request["name"],
             is_active=is_active,
+            product=self.request["product"],
```

Comment:
Or alternatively to the previous comment you can keep cls.plan_name, just give it some uniqueness based on the test class name for example and you will not have to filter by product.